src/brep_faceter.cc

```diff
@@ -189,6 +191,27 @@ void facet_all_volumes(const TopTools_HSequenceOfShape &shape_list,
 
           edge_data edges = make_edge_facets(currentEdge, data);
           mbtool.build_curve(curve, edges, f_vertex_map);
+
+          // add vertices to edges
+          TopTools_IndexedMapOfShape vertices;
+          TopExp::MapShapes(currentEdge, TopAbs_VERTEX, vertices);
+          for (int j = 1; j < vertices.Extent(); j++) {
```

Comment:
is this deliberately ignoring the last element?

OCCT indices tend to go over `[1,n]`, not the normal C style `[0,n)`.

In this case see [NCollection_IndexedMap::FindKey()](https://git.dev.opencascade.org/gitweb/?p=occt.git;a=blob;f=src/NCollection/NCollection_IndexedMap.hxx;h=875068d83c5c3bda49f636134f4017251148e323;hb=HEAD#l372)